src/gurobi_ml/sklearn/pls_regression.py

```diff
@@ -0,0 +1,89 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+""" Module for inserting simple Scikit-Learn PLS regression models into a gurobipy model
+"""
+
+from ..modeling import AbstractPredictorConstr
+from .skgetter import SKgetter
+
+
+def add_pls_regression_constr(gp_model, pls_regression, input_vars, output_vars=None, **kwargs):
+    """Embed pls_regression in gp_model to predict the values of output_vars using input_vars
+
+    Parameters
+    ----------
+    gp_model: :gurobipy:`model`
+        The gurobipy model where the predictor should be inserted.
+    pls_regression: :external+sklearn:py:class:`sklearn.linear_model.LinearRegression`
+     The linear regression to insert. It can be of any of the following types:
+         * :external+sklearn:py:class:`sklearn.cross_decomposition.PLSRegression`
+         * :external+sklearn:py:class:`sklearn.cross_decomposition.PLSCanonical`
+     input_vars: :gurobipy:`mvar` or :gurobipy:`var` array like
+         Decision variables used as input for random forest in model.
+     output_vars: :gurobipy:`mvar` or :gurobipy:`var` array like, optional
+         Decision variables used as output for random forest in model.
+
+    Returns
+    -------
+    PLSRegressionConstr
+        Object containing information about what was added to gp_model to embed the
+        predictor into it
+
+    Note
+    ----
+    |VariablesDimensionsWarn|
+    """
+    return PLSRegressionConstr(gp_model, pls_regression, input_vars, output_vars, **kwargs)
+
+
+class PLSRegressionConstr(SKgetter, AbstractPredictorConstr):
+    """Class to model trained :external+sklearn:py:class:`sklearn.cross_decompostion.PLSRegression` with gurobipy
+
+    Stores the changes to :gurobipy:`model` when embedding an instance into it."""
+
+    def __init__(
+        self,
+        gp_model,
+        predictor,
+        input_vars,
+        output_vars=None,
+        output_type="",
+        **kwargs,
+    ):
+        self.n_outputs_ = 1
+        SKgetter.__init__(self, predictor, output_type, **kwargs)
+        AbstractPredictorConstr.__init__(
+            self,
+            gp_model,
+            input_vars,
+            output_vars,
+            **kwargs,
+        )
+
+    def add_regression_constr(self):
+        """Add the prediction constraints to Gurobi"""
+        x_mean = self.predictor._x_mean
+        x_std = self.predictor._x_std
+        coefs = self.predictor._coef_.T
+        intercept = self.predictor.intercept_
+        self.gp_model.addConstr(
+            self.output == (self.input - x_mean) / x_std @ coefs + intercept,
```

Comment:
```suggestion
            self.output == self.input @ (coefs / x_std[:, np.newaxis]) - x_mean / x_std @ coefs + intercept,
```

We get a runtime warning because the operation is inefficient for gurobipy (it's more efficient to always do 
all numerical operations before combining with matrix variables).
The warning can be removed with the code above (the tests still pass with it, I hope it's correct).

Note that you would also have to import numpy also.

Reply:
Thanks!